File: backtracking/Password.cpp
#include<bits/stdc++.h>
using namespace std;

int l,c;
char arr[20];
char result[20];
int con = 0;
int vo = 0;

void solve(int k, int st){
    if(k==l){
        if(con>=2 && vo>=1){
            for(int i=0;i<l;i++) cout<<result[i];
            cout<<"\n";
            return;
        }
    }

    for(int i=st;i<c;i++) {
        result[k] = arr[i];
        if(arr[i]=='a' || arr[i]=='e' || arr[i]=='i' || arr[i]=='o' || arr[i]=='u') vo++;
        else con++;
        solve(k+1,i+1);
        if(arr[i]=='a' || arr[i]=='e' || arr[i]=='i' || arr[i]=='o' || arr[i]=='u') vo--;
        else con--;
    }
}

int main(void) {
    ios::sync_with_stdio(0);
    cin.tie(0);

    cin>>l>>c;
    for(int i=0;i<c;i++) cin>>arr[i];
    sort(arr, arr+c);
    solve(0,0);
}

File: backtracking/Password.py
import sys
input = sys.stdin.readline

class Password:
    def __init__(self, l, c, arr):
        self.m = l
        self.n = c
        self.arr = sorted(arr)
        self.result = [0] * l
        self.con = 0
        self.vo = 0
    
    def isvowel(self, i):
        return self.arr[i] in 'aeiou'
    
    def solve(self, k, st):
        if k == self.m:
            if self.vo >= 1 and self.con >= 2:
                print(''.join(self.result))
            return
        
        for i in range(st, self.n):
            self.result[k] = self.arr[i]
            if self.isvowel(i): 
                self.vo += 1
            else: 
                self.con += 1
            self.solve(k+1, i+1)
            if self.isvowel(i): 
                self.vo -= 1
            else: 
                self.con -= 1

def main():
    l, c = map(int, input().split())
    arr = input().split()
    PW = Password(l, c, arr)
    PW.solve(0, 0)

if __name__ == "__main__":
    main()
